pgx/src/datum/array.rs

```diff
@@ -505,3 +610,89 @@ where
+                requires_array_brackets: true,
+            }),
+            err @ Err(_) => err,
+        }
+    }
+
+    fn return_sql() -> Result<ReturnVariant, ReturnVariantError> {
+        match T::return_sql() {
+            Ok(ReturnVariant::Plain(SqlVariant::Mapped(sql))) => {
+                Ok(ReturnVariant::Plain(SqlVariant::Mapped(format!("{sql}[]"))))
+            }
+            Ok(ReturnVariant::Plain(SqlVariant::Composite {
+                requires_array_brackets: _,
+            })) => Ok(ReturnVariant::Plain(SqlVariant::Composite {
+                requires_array_brackets: true,
+            })),
+            Ok(ReturnVariant::Plain(SqlVariant::SourceOnly {
+                requires_array_brackets: _,
+            })) => Ok(ReturnVariant::Plain(SqlVariant::SourceOnly {
+                requires_array_brackets: true,
+            })),
+            Ok(ReturnVariant::Plain(SqlVariant::Skip)) => Err(ReturnVariantError::SkipInArray),
+            Ok(ReturnVariant::SetOf(_)) => Err(ReturnVariantError::SetOfInArray),
+            Ok(ReturnVariant::Table(_)) => Err(ReturnVariantError::TableInArray),
+            err @ Err(_) => err,
+        }
+    }
+}
+
+impl<'a, T> SqlTranslatable for VariadicArray<'a, T>
+where
+    T: SqlTranslatable + FromDatum,
+{
+    fn argument_sql() -> Result<SqlVariant, ArgumentError> {
+        match T::argument_sql() {
+            Ok(SqlVariant::Mapped(sql)) => Ok(SqlVariant::Mapped(format!("{sql}[]"))),
+            Ok(SqlVariant::Skip) => Err(ArgumentError::SkipInArray),
+            Ok(SqlVariant::Composite { .. }) => Ok(SqlVariant::Composite {
+                requires_array_brackets: true,
+            }),
+            Ok(SqlVariant::SourceOnly { .. }) => Ok(SqlVariant::SourceOnly {
+                requires_array_brackets: true,
+            }),
+            err @ Err(_) => err,
+        }
+    }
+
+    fn return_sql() -> Result<ReturnVariant, ReturnVariantError> {
+        match T::return_sql() {
+            Ok(ReturnVariant::Plain(SqlVariant::Mapped(sql))) => {
+                Ok(ReturnVariant::Plain(SqlVariant::Mapped(format!("{sql}[]"))))
+            }
+            Ok(ReturnVariant::Plain(SqlVariant::Composite {
+                requires_array_brackets: _,
+            })) => Ok(ReturnVariant::Plain(SqlVariant::Composite {
+                requires_array_brackets: true,
+            })),
+            Ok(ReturnVariant::Plain(SqlVariant::SourceOnly {
+                requires_array_brackets: _,
+            })) => Ok(ReturnVariant::Plain(SqlVariant::SourceOnly {
+                requires_array_brackets: true,
+            })),
+            Ok(ReturnVariant::Plain(SqlVariant::Skip)) => Err(ReturnVariantError::SkipInArray),
+            Ok(ReturnVariant::SetOf(_)) => Err(ReturnVariantError::SetOfInArray),
+            Ok(ReturnVariant::Table(_)) => Err(ReturnVariantError::TableInArray),
+            err @ Err(_) => err,
+        }
+    }
+
+    fn variadic() -> bool {
+        true
+    }
```

Comment:
Why a function and not an associated const?

Reply:
I think it could be! 🤔